### ISSUE TITLE
Ignore React FC for #Cognitive Complexity

### ISSUE THREAD

**Issue by papuruth** [SonarSource/eslint-plugin-sonarjs#226]
**I want to report a bug.**
I have upgraded eslint-plugin-sonarjs to latest version i.e., 0.8.0-125, but still I'm receiving error for cognitive complexity on React FC. An screenshot has been attached.
![image](https://user-images.githubusercontent.com/13766195/122348864-3b662e00-cf69-11eb-96da-69b7dabe09da.png)

**Reproducer**

```javascript
// minimal reproducer if relevant
const RenderContent = (props) => {
  const responsive = {
    // some code here
  };
 // Component code continues
}
```

**Expected behavior**
React FC should be ignored from this rule.

**eslint-plugin-sonarjs version:** 0.8.0-125<br>
**eslint version:** 6.6.0<br>
**Node.js version:** 12.16.1<br>

**Rule key:** S3776

**Comment by vilchik-elena** [comment]
To be clear, React function components are not completely ignored, only there own complexity is counted (excluding nested functions). So I can't tell if you have FP looking at such reproducer.

**Comment by papuruth** [comment]
But in my use case, I see that even the nested functions are being counted. Shouldn't the FCs behave like Class Components?

**Comment by vilchik-elena** [comment]
@papuruth As I said already for FC "there own complexity is counted (excluding nested functions)". If you observe different behaviour please post full reproducer.

**Comment by vilchik-elena** [comment]
FYI there is one fix on the way (#245) so may be that was your problem

**Comment by vilchik-elena** [comment]
closing the issue as no answer from the reporter
